### ISSUE TITLE
Error java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/internal/Platform;

### ISSUE THREAD

**Issue by SumiGhosh** [socketio/socket.io-client-java#369]
I am using HttpLoggingInterceptor lib in my application to see my network logs. But app crashing the point were i add the interceptor. When i comment the line of code, its working. I am pasting few important snippet.

        
`public static OkHttpClient getRequestInteceptor(int state) {
        OkHttpClient.Builder okHttpClient = new OkHttpClient.Builder();
        .
        .
        okHttpClient.addInterceptor(getLoggingInterceptor());
        return okHttpClient.build();
}

private static HttpLoggingInterceptor getLoggingInterceptor(){
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        return logging;
    }
`

**Comment by ltpquang** [comment]
**tl;dr** try to update your `logging-interceptor` to a newer version

Currently the `socket.io` library has a dependency on `okhttp3` version `3.5.0`. In this version, the `Platform` class is located in package `okhttp3.internal.platform`.

Meanwhile, your `logging-interceptor` library has a dependency on `okhttp3` but with an older version than the above. In this older version, the `Platform` class is located in package `okhttp3.internal`.

When you sync your Gradle, because the `socket.io`'s `okhttp` is newer, then it will replace the `logging-interceptor`'s `okhttp`, causing the `logging-interceptor` cannot find the `Platform` class anymore.

**Comment by replyashu** [comment]
Use okHttp3Version = '3.4.1' in gradle, that would resolve the issue, faced this for four hours and resolved like this
